cfme/fixtures/cli.py

```diff
@@ -52,18 +53,28 @@ def fqdn_appliance(appliance):
     else:
         logger.error("Couldn't provision an appliance at all")
         raise SproutException('No provision available')
-    yield apps[0]
+    return apps[0]
 
     apps[0].ssh_client.close()
     sp.destroy_pool(pool_id)
 
 
+@pytest.yield_fixture(scope="function")
+def appliance_with_disk(appliance):
+    yield fqdn_appliance(appliance, preconfigured=False)
+
+
+@pytest.yield_fixture(scope="function")
+def ipa_appliance(appliance):
+    yield fqdn_appliance(appliance, preconfigured=True)
+
+
 @pytest.yield_fixture()
-def ipa_crud(fqdn_appliance, app_creds, ipa_creds):
-    fqdn_appliance.appliance_console_cli.configure_ipa(ipa_creds['ipaserver'],
+def ipa_crud(ipa_appliance, ipa_creds):
+    ipa_appliance.appliance_console_cli.configure_ipa(ipa_creds['ipaserver'],
         ipa_creds['username'], ipa_creds['password'], ipa_creds['domain'], ipa_creds['realm'])
 
-    yield(fqdn_appliance)
+    yield(ipa_appliance)
```

Comment:
**REQUIRED:** I don't see the point in this being a yield fixture as it has no code after the ```yield``` statement